builtin/providers/google/resource_google_project_iam_policy_test.go

```diff
@@ -290,11 +307,56 @@ func testAccCheckGoogleProjectIamPolicyIsMerged(projectRes, policyRes, pid strin
 		}
 
 		// The bindings in both policies should be identical
+		projectP.Bindings = mergeBindings(projectP.Bindings)
+		policyP.Bindings = mergeBindings(policyP.Bindings)
 		sort.Sort(sortableBindings(projectP.Bindings))
 		sort.Sort(sortableBindings(policyP.Bindings))
 		if !reflect.DeepEqual(derefBindings(projectP.Bindings), derefBindings(policyP.Bindings)) {
 			return fmt.Errorf("Project and data source policies do not match: project policy is %+v, data resource policy is  %+v", derefBindings(projectP.Bindings), derefBindings(policyP.Bindings))
 		}
+		return nil
+	}
+}
+
+func testAccCheckGoogleProjectIamPolicyIsMerged(projectRes, policyRes, pid string) resource.TestCheckFunc {
```

Comment:
A lot of the error checking in this function is duplicated from `testAccCheckGoogleProjectIamPolicyExists`, which is called from this fn. I wouldn't mind seeing something like (either or some combination of):
* The error-checking is removed from this function since calling the other one takes care of it
* A helper function could take care of getting the `projectP` and `policyP` values (and doing the error-checking) that gets called from both functions